src/offline/ServiceManager.js

```diff
@@ -0,0 +1,80 @@
+goog.provide('ngeo.offline.ServiceManager');
+
+goog.require('ngeo');
+
+
+ngeo.offline.ServiceManager = class {
+
+  /**
+   * @param {angular.$injector} $injector Main injector.
+   * @struct
+   * @ngInject
+   * @ngdoc service
+   * @ngname ngeoOfflineServiceManager
+   */
+  constructor($injector) {
+
+    /**
+     * @type {angular.$injector}
+     * @private
+     */
+    this.$injector_ = $injector;
+
+    /**
+     * @type {*}
+     * @private
+     */
+    this.saveService_ = null;
+
+    /**
+     * @type {*}
+     * @private
+     */
+    this.restoreService_ = null;
+  }
+
+  /**
+   * Set the service to call on 'save'.
+   * @param {string|null} saveServiceName A service name that can be injected and that have a 'save' method.
+   */
+  setSaveService(saveServiceName) {
+    if (saveServiceName && this.$injector_.has(saveServiceName)) {
+      this.saveService_ = this.$injector_.get(saveServiceName);
+    }   
+  }
+
+  /**
+   * Set the service to call on 'restore'
+   * @param {string|null} restoreServiceName A service name that can be injected and that have a 'restore' method.
+   */
+  setRestoreService(restoreServiceName) {
+    if (restoreServiceName && this.$injector_.has(restoreServiceName)) {
+      this.restoreService_ = this.$injector_.get(restoreServiceName);
+    } 
+  }
+
+  /**
+   * Ask the provided service to save the data to an offline purpose
+   */
+  save() {
+    if (this.saveService_) {
```

Comment:
Would be good for devs to have a warning if you call this method while not saveService is declared ?

Reply:
Indeed